### PR TITLE
Adding fakes

### PR DESCRIPTION
**Pull request by tommy9**
Adds 14 new fakes and stubs from the list in issue #2891 

Also changes the way the native function calls are made when setting the PassThrough property to true. EasyHook has a way to call the original function which saves having to define the call signature separately for VBA6.dll, VBE6.dll and VBE7.dll. It is also more reliable as some functions look for calling project information which works better when EasyHook reroutes the call instead of making a new call from .Net.

The only remaining procedures in the original list are now:

- Name (seems to use function __vbaNameFile but this is not exported and getting an error when trying to hook)
- Move (don't know what this refers to, guess it is in the list by mistake?)
- Various file functions that include a file number (FreeFile, Open, FileAttr, EOF, Loc, LOF, Reset, Seek, Unlock and Width) as wasn't sure about usefulness of hooking and being unable to find all the dll function names matching them
- The pointer functions (VarPtr, ObjPtr, StrPtr)

I can try adding more if there is interest in any of the above. I'll also submit a separate pull request with a wiki update to cover faking.

### CODE REVIEW

File: Rubberduck.Main/ComClientLibrary/Abstract/UnitTesting/IFakesProvider.cs [hunk 1]
```diff
@@ -79,5 +79,61 @@ public interface IFakesProvider
         [DispId(17)]
         [Description("Configures VBA.DateTime.Date calls.")]
         IFake Date { get; }
+
+        [DispId(18)]
+        [Description("Configures VBA.Math.Rnd calls.")]
+        IFake Rnd { get; }
+
+        [DispId(19)]
+        [Description("Configures VBA.Iteraction.DeleteSetting calls.")]
```
Comment (Vogel612) [inline Rubberduck.Main/ComClientLibrary/Abstract/UnitTesting/IFakesProvider.cs:88]: typo: Interaction (same for `SaveSetting`, `GetSetting` and `GetAllSettings`)

File: Rubberduck.Main/ComClientLibrary/UnitTesting/Fakes/IMEStatus.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        [UnmanagedFunctionPointer(CallingConvention.StdCall, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.I2)]
+        private delegate short IMEStatusDelegate();
+
+        public short IMEStatusCallback()
+        {
+            OnCallBack(true);
+            if (!TrySetReturnValue())                          // specific invocation
+            {
+                TrySetReturnValue(true);                       // any invocation
```
Comment (Vogel612) [inline Rubberduck.Main/ComClientLibrary/UnitTesting/Fakes/IMEStatus.cs:24]: not a fan of the end-of-line comments

**Comment by tommy9** [comment]
Hello. It seems I can't create a pull request for the edits to the wiki. I've updated the sidebar and unit testing pages which can be seen on my copy here:

https://raw.githubusercontent.com/wiki/tommy9/Rubberduck/_Sidebar.md
https://github.com/tommy9/Rubberduck/wiki/Unit-Testing

If these are okay, could someone please copy to the original wiki or give me write access?

**Comment by retailcoder** [comment]
Hi @tommy9, I've sent an invite your way; the wiki was freely editable for a long time, but a little while ago we started receiving some spammy edits and reluctantly locked it. Let me know if the "contributors" team grants you wiki edit privs, if not I'll bump you to "dev". Cheers!

File: Rubberduck.Main/ComClientLibrary/Abstract/UnitTesting/IFakesProvider.cs (hunk 1 above)
Comment (tommy9) [reply Rubberduck.Main/ComClientLibrary/Abstract/UnitTesting/IFakesProvider.cs]: Good catch, sending a fix

File: Rubberduck.Main/ComClientLibrary/UnitTesting/Fakes/IMEStatus.cs (hunk 2 above)
Comment (tommy9) [reply Rubberduck.Main/ComClientLibrary/UnitTesting/Fakes/IMEStatus.cs]: Thanks, cleaning this and a few others

**Review by Vogel612 — APPROVED** [review]

**Review by retailcoder — APPROVED** [review]
Thanks for this awesome contribution, and nice clean-up!

**Pull request merged**